LION/optimizers/LIONsolver.py

```diff
@@ -141,46 +155,77 @@ def set_validation(
         """
         self.validation_loader = validation_loader
         self.validation_freq = validation_freq
-        self.validation_fn = validation_fn if validation_fn is not None else self.loss_fn
+        self.validation_fn = (
+            validation_fn if validation_fn is not None else self.loss_fn
+        )
         self.validation_fname = validation_fname
 
-    def set_testing(self, test_loader: DataLoader, testing_fn: Callable):
+    def set_testing(
+        self, test_loader: DataLoader, testing_fn: Optional[Callable] = None
+    ):
         """
         This function sets the testing data
         """
         self.test_loader = test_loader
-        self.testing_fn = testing_fn
+        self.testing_fn = testing_fn if testing_fn is not None else self.loss_fn
+
+    def set_saving(self, save_folder: str | pathlib.Path, final_result_fname: str):
+        if isinstance(save_folder, str):
+            save_folder = pathlib.Path(save_folder)
+        if not save_folder.is_dir():
+            raise ValueError(
+                f"Save folder '{save_folder}' is not a directory, failed to set saving."
+            )
+
+        self.save_folder = save_folder
+        self.final_result_fname = final_result_fname
+
+    def set_loading(self, load_folder: str | pathlib.Path, do_load: bool = False):
+        if isinstance(load_folder, str):
+            load_folder = pathlib.Path(load_folder)
+        if not load_folder.is_dir():
+            raise ValueError(
+                f"Save folder '{load_folder}' is not a directory, failed to set saving."
+            )
+
+        self.load_folder = load_folder
+        self.do_load_checkpoint = do_load
```

Comment:
We were going to change this. Is this here because you haven't merged main here yet?

Reply:
So do you want to just entirely do away with set_loading and have set_checkpointing have a signature like (checkpoint fname, checkpoint_freq, do_load, save_folder, load_folder)?

Reply:
yup! do it that way, then I will fiddle with it later on if needed